From d293a839cbcee0481250fbaf18ee6588dd05c964 Mon Sep 17 00:00:00 2001
From: Adam Ford <aford173@gmail.com>
Date: Mon, 17 Aug 2020 11:19:23 -0500
Subject: [PATCH 7/7] beacon-rzg2m: Configure RAMdisk scripts for booting

The board can boot a RAMdisk, but instead of manually typing
a bunch of instructions, this patch adds a series of variables.

To boot from RAMdisk,  run mmcramboot

Signed-off-by: Adam Ford <aford173@gmail.com>
---
 include/configs/beacon-rzg2m.h | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/include/configs/beacon-rzg2m.h b/include/configs/beacon-rzg2m.h
index 3d51efa195..96b1d1fe4c 100644
--- a/include/configs/beacon-rzg2m.h
+++ b/include/configs/beacon-rzg2m.h
@@ -36,6 +36,7 @@
 	"fdt_file=" CONFIG_DEFAULT_FDT_FILE "\0" \
 	"initrd_addr=0x43800000\0"		\
 	"initrd_high=0xffffffffffffffff\0" \
+	"ramdisk_file=rootfs.cpio.uboot\0" \
 	"mmcdev=0\0" \
 	"mmcpart=1\0" \
 	"mmcrootpart=2\0" \
@@ -43,11 +44,18 @@
 	"mmcautodetect=yes\0" \
 	"mmcargs=setenv bootargs console=${console} " \
 	" root=PARTUUID=${uuid} rootwait rw ${optargs}\0" \
+	"ramargs=setenv bootargs console=${console} " \
+	" root=/dev/ram rw ${optargs}\0"                    \
 	"loadbootscript=load mmc ${mmcdev}:${mmcpart} ${loadaddr} ${script};\0" \
 	"bootscript=echo Running bootscript from mmc ...; " \
 		"source\0" \
 	"loadimage=load mmc ${mmcdev}:${mmcpart} ${loadaddr} ${image}\0" \
 	"loadfdt=load mmc ${mmcdev}:${mmcpart} ${fdt_addr} ${fdt_file}\0" \
+	"loadramdisk=load mmc ${mmcdev}:${mmcpart} ${initrd_addr} "\
+	" ${ramdisk_file}\0" \
+	"mmcramboot=run ramargs; run loadimage;" \
+	" run loadfdt; run loadramdisk;" \
+	" booti ${loadaddr} ${initrd_addr} ${fdt_addr}\0" \
 	"mmcboot=echo Booting from mmc ...; " \
 		"run finduuid; run mmcargs; " \
 		"if test ${boot_fdt} = yes || test ${boot_fdt} = try; then " \
-- 
2.17.1

